language-plutus-core/src/Language/PlutusCore/Name.hs

```diff
@@ -76,6 +78,19 @@ newtype Unique = Unique { unUnique :: Int }
     deriving (Eq, Show, Ord, Lift)
     deriving newtype (NFData)
 
+class HasUnique a where
+    unique :: Lens' a Unique
+
+instance HasUnique (Name a) where
```

Comment:
Are these instances for `Name` and `TyName` used anywhere?

Reply:
Well, you need them if you want to rename a `Term TyName Name`! Which was the whole point - but I forgot to add the actual use in here, I'll include that.